htdocs/web_portal/controllers/downtime/add_downtime.php

```diff
@@ -158,6 +158,21 @@ function draw(\User $user = null) {
         die(json_encode(array('UTC', 0)));
     }
 
+    // URL Mapping
+    // If the user wants to fetch a specific service endpoints, show only that service endpoints.
+    else if(isset($_GET['site']) && isset($_GET['se'])) {
+        $se = \Factory::getServiceService()->getService($_GET['se']);
+        $site = \Factory::getSiteService()->getSite($_GET['site']);
+        if(\Factory::getRoleActionAuthorisationService()->authoriseAction(\Action::EDIT_OBJECT, $site, $user)->getGrantAction() == FALSE){
+            throw new \Exception("You don't have permission over $site");
```

Comment:
```suggestion
            throw new \Exception("You don't have permissions over $site");
```

makes more grammatical sense?

Reply:
im not sure, permissions seems to roll off the tongue better

Reply:
Based on the code above, it is authorized only on the service ID and site level. So, If you think it is better, I am open and happy for the discussion, and we can create a tech debt for it to discuss and resolve.

Reply:
The reason I am saying is because we might need to change across the codebase.